### PR TITLE
Add StateMetadataFeature to bindings

### LINKED ISSUE #1650: Add StateMetadataFeature
https://github.com/iotaledger/tips/blob/tip54/tips/TIP-0054/tip-0054.md#state-metadata-feature

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add StateMetadataFeature to bindings
Update MetadataFeature in JS
Enable anchor state metadata validation

## Links to any relevant issues

Fixes #1650 

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Running the feature examples in nodejs and python

### CODE REVIEW

File: bindings/python/iota_sdk/types/feature.py [hunk 1]
```diff
@@ -81,6 +81,20 @@ class MetadataFeature:
     entries: Dict[str, HexStr]
 
 
+@json
+@dataclass
+class StateMetadataFeature:
+    """Defines metadata, arbitrary binary data, that will be stored in the output.
```
Comment (thibault-martinez) [inline bindings/python/iota_sdk/types/feature.py]: I know we already had that in Rust but is there no better doc than that somewhere? Maybe in the TIP?

Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/feature.py]: `A Metadata Feature that can only be changed by the State Controller.` want this?

Comment (thibault-martinez) [reply bindings/python/iota_sdk/types/feature.py]: That's not exceptional but at least it's a bit more accurate yeah

File: bindings/nodejs/examples/how_tos/outputs/features.ts [hunk 2]
```diff
@@ -53,15 +53,17 @@ async function run() {
...
+            ],
         });
 
         // Output with immutable metadata feature
         const nftOutputWithImmutableMetadata = await client.buildNftOutput({
             nftId: '0x0000000000000000000000000000000000000000000000000000000000000000',
             unlockConditions: [addressUnlockCondition],
             immutableFeatures: [
-                new MetadataFeature(utf8ToHex('Hello, World!')),
+                new MetadataFeature({ data: utf8ToHex('Hello, World!') }),
```
Comment (thibault-martinez) [inline bindings/nodejs/examples/how_tos/outputs/features.ts:66]: Is this correct? No quotes on `data`?

Comment (Thoralf-M) [reply bindings/nodejs/examples/how_tos/outputs/features.ts:66]: Yes, I even added the quotes initially but they get removed from format

Comment (thibault-martinez) [reply bindings/nodejs/examples/how_tos/outputs/features.ts:66]: That's so weird, it feels wrong. Did you try it by any chance?

Comment (Thoralf-M) [reply bindings/nodejs/examples/how_tos/outputs/features.ts:66]: That's normal in JS world `console.log(JSON.stringify({data: "0x..."}))` ->`'{"data":"0x..."}'`

Comment (Thoralf-M) [reply bindings/nodejs/examples/how_tos/outputs/features.ts:66]: Also tried the example `Running the feature examples in nodejs and python` with the feature

Comment (thibault-martinez) [reply bindings/nodejs/examples/how_tos/outputs/features.ts:66]: Alright, I guess I'll never understand JS

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**